tools/python/mwm/dump_mwm.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python
+import sys, os.path, random
+import json
+from mwm import MWM
+
+if len(sys.argv) < 2:
+    print('Dumps some MWM structures.')
+    print('Usage: {0} <country.mwm>'.format(sys.argv[0]))
+    sys.exit(1)
+
+mwm = MWM(open(sys.argv[1], 'rb'))
+mwm.read_types(os.path.join(os.path.dirname(sys.argv[0]), '..', '..', '..', 'data', 'types.txt'))
+print('Tags:')
+tvv = sorted([(k, v[0], v[1]) for k, v in mwm.tags.items()], key=lambda x: x[1])
+for tv in tvv:
+    print('  {0:<8}: offs {1:9} len {2:8}'.format(tv[0], tv[1], tv[2]))
+v = mwm.read_version()
+print('Format: {0}, version: {1}'.format(v['fmt'], v['version'].strftime('%Y-%m-%d %H:%M')))
+print('Header: {0}'.format(mwm.read_header()))
+print('Metadata count: {0}'.format(len(mwm.read_metadata())))
+
+cross = mwm.read_crossmwm()
+if cross:
+    print('Outgoing points: {0}, incoming: {1}'.format(len(cross['out']), len(cross['in'])))
+    print('Outgoing regions: {0}'.format(set(cross['neighbours'])))
+
+print('Sample features:')
+# Print 5 random features ~500 features apart
```

Comment:
So, why not to set probability to, say, 0.999 instead of 0.0001? How have
you selected this magic number?

Whatever, if you need to select exactly 5 random features, please, do it
correctly. Your algorithm is highly biased towards the head of the features
list. If you don't know how to do that properly, contact me or @mpimenov.
Or do some math by your self, it's a well known problem.
06 июня 2016 г. 18:07 пользователь "Ilya Zverev" <notifications@github.com>
написал:

> In tools/python/mwm/dump_mwm.py
> <https://github.com/mapsme/omim/pull/3446#discussion_r65907237>:
>
> > +print('Tags:')
> > +tvv = sorted([(k, v[0], v[1]) for k, v in mwm.tags.items()], key=lambda x: x[1])
> > +for tv in tvv:
> > +    print('  {0:<8}: offs {1:9} len {2:8}'.format(tv[0], tv[1], tv[2]))
> > +v = mwm.read_version()
> > +print('Format: {0}, version: {1}'.format(v['fmt'], v['version'].strftime('%Y-%m-%d %H:%M')))
> > +print('Header: {0}'.format(mwm.read_header()))
> > +print('Metadata count: {0}'.format(len(mwm.read_metadata())))
> > +
> > +cross = mwm.read_crossmwm()
> > +if cross:
> > +    print('Outgoing points: {0}, incoming: {1}'.format(len(cross['out']), len(cross['in'])))
> > +    print('Outgoing regions: {0}'.format(set(cross['neighbours'])))
> > +
> > +print('Sample features:')
> > +# Print 5 random features ~500 features apart
>
> To deem an algorithm correct or incorrect, you need to know the exact
> problem it is solving. In this case I need whatever 5 random elements, and
> I obviously get them.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/3446/files/92f7ae7c5f2b6e0b61da2e311a0c5e16cbf1e3e3#r65907237>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAbvp67G5FU9nEnb33qZiCXov6c4cjWQks5qJDeagaJpZM4IuHLu>
> .
>